### PR TITLE
Create a script to collect all bufmod TODOs

### PR DESCRIPTION
**Pull request by doriable**
This is a script for collecting all `TODO`s on `bufmod` that are
not on `main`. Sample output:

```
private/bufpkg/bufmodule/remote_dep.go: 44 // TODO: This needs a LOT of testing.
private/pkg/connectclient/connectclient.go: 21 // TODO: eliminate config, just make the below MakeOptions, add httpClient as a parameter to Make
private/pkg/dag/graph.go: 45 // TODO: It really stinks that we have to use the constructor. We have what amounts
private/pkg/slicesext/slicesext.go: 146 // TODO: Delete this in favor of slices.Clone.
private/pkg/slicesext/slicesext.go: 166 // TODO: Make ToStructMap use this logic, remove ToStructMapOmitEmpty, to match other functions.
private/pkg/slicesext/slicesext.go: 315 // TODO: Replace with slices.Sort when we only support Go versions >= 1.21.
private/pkg/syncext/syncext.go: 25 // TODO: This is directly copied from 1.21 source, remove when no longer need <1.20.
private/pkg/syncext/syncext.go: 57 // TODO: This is directly copied from 1.21 source, remove when no longer need <1.20.
```

### CODE REVIEW

**Comment by doriable** [comment]
@bufdev I wasn't sure if we wanted to put this as a part of `make generate`, I didn't want to inadvertently cause CI issues by doing that. It seems reasonable to generate and/use on an ad-hoc basis for now but yeah, let me know what you think.

File: make/buf/scripts/newtodos.bash [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  f=="?" {if (/^\+\+\+ /) f=substr($0, 7)"\n"; next}
+  /^@@/ {n=$3; sub(/,.*/,"",n); n=0+$3; next}
+  /^\+.*TODO/ {print f n ":" substr($0,2); f=""}
+  substr($0,1,1)~/[ +]/ {n++}')"
+
+FILENAME=""
+while read -r line; do
+  if [[ "${line}" =~ [a-z,0-9]+.go ]]; then
+    FILENAME="${line}"
+  elif [[ "${line}" =~ PACKAGES.md ]]; then
```
Comment (bufdev) [inline make/buf/scripts/newtodos.bash:23]: There's no reason to ignore `PACKAGES.md`, if someone wants to ignore this, you can grep it out after the fact

File: make/buf/scripts/newtodos.bash [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+TODOS="$(git diff "${BASE_BRANCH}" | awk '
+  /^diff / {f="?"; next}
+  f=="?" {if (/^\+\+\+ /) f=substr($0, 7)"\n"; next}
+  /^@@/ {n=$3; sub(/,.*/,"",n); n=0+$3; next}
+  /^\+.*TODO/ {print f n ":" substr($0,2); f=""}
+  substr($0,1,1)~/[ +]/ {n++}')"
+
+FILENAME=""
+while read -r line; do
+  if [[ "${line}" =~ [a-z,0-9]+.go ]]; then
```
Comment (bufdev) [inline make/buf/scripts/newtodos.bash:21]: Optimally, we want all TODOS, not just Golang

**Comment by bufdev** [comment]
This script still has bugs, example:

```
586:					// TODO: sweep and check paths gone: 593 // TODO: add default values
586:					// TODO: sweep and check paths gone: 773 // TODO in v2
```

File: make/buf/scripts/newtodos.bash (hunk 2 above)
Comment (doriable) [reply make/buf/scripts/newtodos.bash]: Fixed this up, I did exclude this script file though.

**Review by bufdev — APPROVED** [review]

**Pull request merged**